extension/app/ts/simulation/services/ETHSubscriptionService.ts

```diff
@@ -0,0 +1,92 @@
+import { EthSubscribeParams, JsonRpcMessage, JsonRpcNewHeadsNotification } from '../../utils/wire-types.js'
+import { ErrorWithData } from '../../utils/errors.js'
+import { Future } from '../../utils/future.js'
+
+type Subscription = {
+	callback: (subscriptionId: string, reply: JsonRpcNewHeadsNotification) => void
+	params: EthSubscribeParams,
+	rpcSocket: WebSocket
+}
+
+export class ETHSubscriptionService {
```

Comment:
To match naming convention elsewhere:
```suggestion
export class EthereumSubscriptionService {
```
Alternatively (not a fan of this but better than `ETH`):
```suggestion
export class EthSubscriptionService {
```

Reply:
the point was to refer to `eth_subscription` with the naming

Reply:
I recommend not actually implementing this service using `eth_subscription` under the hood, hence I recommend not porting that name over to the class.  😀